Adding Json to FieldType and missing helper methods

mchlgibs opened the pull request:
Adding Json as a field type

Complete list is here: https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#TableFieldSchema

Json seems to be the only one that was missing in the enum, but Json & Geography were both missing in the helper functions.

This should fix https://github.com/lquerel/gcp-bigquery-client/issues/56

mchlgibs commented:
I haven't checked. I'll look at it over the weekend and see what I can do about adding tests. (And code if necessary.)

mchlgibs commented:
Checked and added a test.

mchlgibs commented:
@lquerel it's ready for you to inspect and merge.

lquerel commented:
Sorry for the delay. 

In your example the Rust type of the json column is String but in my opinion it will be more natural to use serde_json::value::Value in order to deserialize the json value in a dynamic structure directly navigable.

mchlgibs commented:
I added a third commit, where I change the type of the json column to serde_json::value::Value. There was kind of a trick though - when you serialize it into the request - you have to serialize that column as a single string, rather than as a JSON object. So I implemented a function to do that and updated the test. Assuming we add this functionality, other users will have to annotate their structs similarly.

I'd appreciate advice on the following:
1) In the helper method, I do an ``unwrap()`` - I feel like I shouldn't be, but I couldn't get the error type conversions to work. Specifically, this occurs in the case where a valid serde_json::value::Value object fails to convert to a string.
2) Is the helper in the right place, or is there a more logical place to put it?
3) Is this even the right way to go?

mchlgibs commented:
@lquerel - all done :)

Assuming you like this, I'd recommend a squash, rather than just a merge.

lquerel commented:
> @lquerel - all done :)
> 
> Assuming you like this, I'd recommend a squash, rather than just a merge.

I will work on this over the weekend. Thanks

lquerel commented:
@mchlgibs merged and published. Thank you.

mchlgibs commented:
Thank you so much.
